pkg/uefi/nvram.go

```diff
@@ -0,0 +1,475 @@
+// Copyright 2019 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// NVAR decoding logic ported from UEFITool Copyright (c) 2016, Nikolaj Schlej.
+
+package uefi
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"unicode/utf16"
+
+	"github.com/linuxboot/fiano/pkg/guid"
+)
+
+// NVarAttribute represent Attributes
+type NVarAttribute uint8
+
+// Attributes
+const (
+	NVarEntryRuntime       NVarAttribute = 0x01
```

Comment:
Is there a spec or reference implementation where this is defined, or has the whole thing been reverse engineered?

Reply:
Trammel Hudson have the same here: https://trmm.net/UEFI_notes
chipsec also: https://github.com/chipsec/chipsec/blob/master/chipsec/hal/uefi_platform.py#L143

Reply:
Probably found a reference to the source, it's definitely reverse engineered
https://firmwaresecurity.com/2016/04/16/nikolaj-on-nvram-formats-part-3/
Pointing to the blog of Nikolaj Schlej in russian:
https://habr.com/en/post/281901/

Reply:
Thanks for digging up these links!

Reply:
done